### DISCUSSION TITLE
[Chapter 02] 설정파일과 POJO

### DISCUSSION THREAD

**Discussion by BvrPark** [cowaine/spring-boot-for-msa#7] [Chapter 02. Framework and Spring Boot] [answered]
### 1. Maven vs Gradle
책의 예제에서는 Maven 설정파일을 사용하고 있는데 실제로 다른 사람들의 repository를 들어가면 Gradle을 많이 봤습니다.
Maven이 xml파일을 사용해서 스크립트 파일을 사용하는 Gradle에 비해서 가독성(?)이 떨어진다? 
이것 말고 둘의 차이점이 또 있을까요...?
제가 다른 분들 github를 봤을 땐 Gradle을 더 많이 봐서 뭔가 또 다른 차이점이 있는지 궁금합니다.

### 2. POJO
POJO의 개념을 순수한 자바객체로 알고있습니다.
그래서 프레임워크나 라이브러리에 종속되지 않는다고 알고 있는데

1. 프레임워크나 라이브러리를 쓰면 더 편리할 것 같은데 POJO는 어떤 상황에서 필요한지 궁금합니다.
2. 1번 질문과 관련해서 그러면 POJO의 장점?은 뭐가 있는지도 궁금합니다.

**Comment by corock** [comment, marked as answer]
1. 스프링만 예로 들어도 _root-context_ 나 _servlet-context_ 등의 XML 파일을 걷어내고 _Java Config_ 를 사용하는 추세인 것처럼
_pom.xml_ 도 결국 XML 이기에 _groovy_ 나 코틀린 DSL 을 대신 사용하려고 하지 않나 싶습니다.
그리고 안드로이드 모바일 앱의 기본 빌드 도구가 _Gradle_ 이라 다양한 디바이스에 대한 호환성도 고려해 보면 Gradle 이 좋긴 한 것 같습니다.
(_cf._ [Gradle vs Maven Comparison](https://gradle.org/maven-vs-gradle))

2. 편리성을 취하는 대신 프레임워크 또는 라이브러리에 대한 의존도가 높아지지 않을까요?
기술은 계속해서 변하고 언젠가는 사용하고 있는 기술이 레거시가 됐을 때, 그땐 새로운 기술로 전환하고 싶어도 못 하는 상황이 올 수 있을 것 같아요.
